controllers/dependencies.go

```diff
@@ -97,7 +96,7 @@ func (r *OperatorPipelineReconciler) applyManifests(fileName string, Namespace s
 	}
 
 	obj.SetNamespace(Namespace)
-	if err = r.Client.Create(context.Background(), obj); err != nil {
+	if err = r.Client.Create(context.Background(), obj); err != nil && errors.IsNotFound(err) {
```

Comment:
This has the risk that other errors (bad requests/conflict) will not be handled properly this way. IMO, it is better to ignore `IsAlreadyExist` error and let all the other errors to fail the method call. I mean:
```
if err = r.Client.Create(context.Background(), obj); err != nil && !errors.IsAlreadyExists(err) {
```
Ref: https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsAlreadyExists

Reply:
@samira-barouti This is the error I was looking for, thanks! Will update and test with this one as well.

Reply:
Ended up going a different route and updating the resource if it already exists in the cluster, since we clone the repo each reconciliation, and that could have more pipelines/tasks.